### ISSUE TITLE
Detect Hub vs Binder in launch button

### ISSUE THREAD

**Issue by agoose77** [jupyter-book/myst-theme#507]
We can programmatically detect the presence of a Hub vs Binder from querying API endpoints in the URL.

**Comment by agoose77** [comment]
The canonical relative URLs that indicate a Hub vs Binder are:

- GET `hub/api/`
- GET `versions`

Iff. a Hub and BinderHub deployment share the same handler namespace, we should treat the result as a BinderHub.

**Comment by stevejpurves** [comment]
@agoose77 yes that `*` allow origins change/assumption is a bit of an antipattern? it'll be down to individual hubs to allow the consumers they want to let in and there is already a config surface in jhub for that (as @manics points out).

The way around that for non static sites is to expose a new route that can take the request server-side, but that doesn't help at all for static deployments.

There are no other get requests at all that allow a binderhub to be differentiated from a jupyterhub? mind you isn't having the explicit type set in `project.services.xxxx` a better/acceptable route forward anyways?

**Comment by agoose77** [comment]
@stevejpurves My proposal is to add it to the unauthenticated routes like `/hub/api/` etc., not the top-level!

**Comment by manics** [comment]
I don't think we should override admin's preferences for which external sites can interact with their hub, even for unauthenticated endpoints.

**Comment by agoose77** [comment]
@manics how strict are you feelings on this? For example, is `GET ${HUB_URL}/hub/api/` something that you wouldn't be comfortable exposing by default?

**Comment by agoose77** [comment]
Furthermore, I want to be clear -- we should not be overriding preferences in the sense that admins should be able to disable this.

**Comment by agoose77** [comment]
@stevejpurves 
> it'll be down to individual hubs to allow the consumers they want to let in

The issue with the current approach is that in-practice it means that most hubs don't support even the simplest interrogation of "what are you?". It's hard to author a useful feature if by-default it is broken out of the box.

> mind you isn't having the explicit type set in project.services.xxxx a better/acceptable route forward anyways?

This is about bring-your-own-hub, whereby we don't know what the service is until the user provides it.
